src/main/java/frc/robot/VisionProcessor.java

```diff
@@ -132,19 +145,34 @@ public void process(Mat sourceFrame) {
         ArrayList<RotatedRect> initial = minimumBoundingRectangle(parent.filterContoursOutput());
         ArrayList<RotatedRect> findContours = minimumBoundingRectangle(parent.findContoursOutput());
         ArrayList<RotatedRect> ok = new GibberishRectangleFilter().filter(initial);
-        ArrayList<RotatedRect> selected = new DumbAndAloneRectangleFilter().filter(ok);
-
+        numRects = ok.size();
+
+        //reporter.getTargetAlignButtonPressed() == false || prevNumRects != numRects || frames.getTicks()<10 
+        //is the target lock logic, if all of these are false than target lock is initiated
+        if(reporter.getTargetAlignButtonPressed() == false || prevNumRects != numRects || frames.getTicks()<10){
```

Comment:
why do we trigger with ticks <10? it seems like we'll unnecessarily do the logic the first 9 times.?